### PR TITLE
Feature/l3 239

### PR DESCRIPTION
**Pull request by n3op2**
# What

A cluster update. Adding ingress controller along with the class for each persona and stored in nginx folder. More on this -> https://digicatapult.atlassian.net/browse/L3-239

### CODE REVIEW

File: clusters/l3-sqnc/capacity/nginx/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - certificate.yaml
+  - release.yaml
+  - source.yaml
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/kustomization.yaml:6]: Remove this.

File: clusters/l3-sqnc/capacity/nginx/source.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: bitnami
+  namespace: capacity
+spec:
+  interval: 10m
+  url: https://charts.bitnami.com/bitnami
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/source.yaml:8]: Move this into the Capacity level `source.yaml`

File: clusters/l3-sqnc/capacity/nginx/values.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+watchIngressWithoutClass: false
+scope:
+  enabled: true
+ingressClassResource:
+  name: nginx-capacity
+  default: false
+config:
+    global-auth-url: "http://oathkeeper-api.ory.svc.cluster.local/decisions$request_uri"
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:8]: Remove, we have not set this up yet.

File: clusters/l3-sqnc/capacity/nginx/values.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+service:
+  type: LoadBalancer
+  annotations: 
+    service.beta.kubernetes.io/azure-load-balancer-health-probe-request-path: /healthz
+    external-dns.alpha.kubernetes.io/hostname: capacity.dsch-l3.com
+    cert-manager.io/cluster-issuer: letsencrypt-prod
+    cert-manager.io/common-name: capacity.dsch-l3.com
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:15]: Remove as cert-manager is not yet installed.

File: clusters/l3-sqnc/capacity/nginx/values.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+metrics:
+  enabled: true
+  serviceMonitor:
+    enabled: true
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:19]: Remove as metrics are not yet installed.

File: clusters/l3-sqnc/capacity/nginx/values.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+extraArgs:
+  default-ssl-certificate: "capacity/capacity-dsch-l3-com-prod-tls"
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:21]: Remove as cert-manager is not yet installed.

File: clusters/l3-sqnc/capacity/nginx/values.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+    operator: Equal
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/control-plane
+    operator: Equal
+containerPorts:
+  http: 80
+  https: 443
+  metrics: 10254
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:34]: Remove.

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]
You need to include `nginx` in `resources` in `capacity's` `kustomization.yaml` and also tell it to include the `values-nginx` and to get it from `nginx/values.yaml`

File: clusters/l3-sqnc/capacity/nginx/kustomization.yaml (hunk 1 above)
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/kustomization.yaml:6]:
```suggestion
```

File: clusters/l3-sqnc/capacity/nginx/values.yaml (hunk 7 above)
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/capacity/nginx/values.yaml:34]:
```suggestion
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: clusters/l3-sqnc/capacity/nginx/values.yaml (hunk 3 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/values.yaml]: should be removed

File: clusters/l3-sqnc/capacity/nginx/values.yaml (hunk 4 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/values.yaml]: this as well. other that required removal will mark as resolved with a comment.

File: clusters/l3-sqnc/capacity/nginx/values.yaml (hunk 6 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/values.yaml]: removed along with cert-amanger.

File: clusters/l3-sqnc/capacity/nginx/values.yaml (hunk 5 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/values.yaml]: done

File: clusters/l3-sqnc/capacity/nginx/source.yaml (hunk 2 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/source.yaml]: This should be updated along with the values file. When you get a moment please take a look at the latest.

File: clusters/l3-sqnc/optimiser/nginx/values.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+    operator: Equal
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/control-plane
+    operator: Equal
+containerPorts:
+  http: 80
+  https: 443
+  metrics: 10254
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/optimiser/nginx/values.yaml:24]:
```suggestion
```

File: clusters/l3-sqnc/order/nginx/kustomization.yaml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - certificate.yaml
+  - release.yaml
+  - source.yaml
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/order/nginx/kustomization.yaml:6]:
```suggestion
```

File: clusters/l3-sqnc/order/nginx/values.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+    operator: Equal
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/control-plane
+    operator: Equal
+containerPorts:
+  http: 80
+  https: 443
+  metrics: 10254
```
Comment (dblane-digicatapult) [inline clusters/l3-sqnc/order/nginx/values.yaml:24]:
```suggestion
```

**Review by dblane-digicatapult — CHANGES_REQUESTED** [review]

File: clusters/l3-sqnc/capacity/nginx/release.yaml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+      retries: -1
+  releaseName: nginx-ingress-controller
+  chart:
+    spec:
+      chart: nginx-ingress-controller
+      sourceRef:
+        kind: HelmRepository
+        name: bitnami
+  interval: 10m0s
+  # Default values
```
Comment (Ellenn-A) [inline clusters/l3-sqnc/capacity/nginx/release.yaml:19]: is this meant to stay in?

**Review by dblane-digicatapult — APPROVED** [review]

File: clusters/l3-sqnc/capacity/nginx/release.yaml (hunk 11 above)
Comment (n3op2) [reply clusters/l3-sqnc/capacity/nginx/release.yaml:19]: Yea, Dave just confirmed (being in the office perks)

**Pull request merged**